wordpress/wp-content/plugins/memberful-wp/src/acl.php

```diff
@@ -195,7 +214,54 @@ function memberful_wp_extract_slug_ids_and_user($args) {
  * @param integer $post_id ID of the post that should have access checked
  */
 function memberful_can_user_access_post( $user, $post ) {
-  $restricted_posts = memberful_wp_user_disallowed_post_ids( $user );
+  if ( !empty( memberful_terms_restricting_post( $user, $post ))) {
```

Comment:
@northerner I just started digging into the ACL code, but I have a feeling that we shouldn't modify `memberful_can_user_access_post()`. Instead of that we should add term restrictions checks into `memberful_wp_user_disallowed_ids_from_acl()` which is used by `memberful_wp_user_disallowed_post_ids()`.

That should work nicely with merging restrictions settings from posts and terms as we agreed on it in Basecamp.